source/memory.tex

```diff
@@ -484,77 +487,78 @@
+    constexpr shared_ptr<T> make_shared(size_t N);                              // \tcode{T} is \tcode{U[]}
   template<class T, class A>
-    shared_ptr<T> allocate_shared(const A& a, size_t N);                        // \tcode{T} is \tcode{U[]}
+    constexpr shared_ptr<T> allocate_shared(const A& a, size_t N);              // \tcode{T} is \tcode{U[]}
 
   template<class T>
-    shared_ptr<T> make_shared();                                                // \tcode{T} is \tcode{U[N]}
+    constexpr shared_ptr<T> make_shared();                                      // \tcode{T} is \tcode{U[N]}
   template<class T, class A>
-    shared_ptr<T> allocate_shared(const A& a);                                  // \tcode{T} is \tcode{U[N]}
+    constexpr shared_ptr<T> allocate_shared(const A& a);                        // \tcode{T} is \tcode{U[N]}
 
   template<class T>
-    shared_ptr<T> make_shared(size_t N, const remove_extent_t<T>& u);           // \tcode{T} is \tcode{U[]}
+    constexpr shared_ptr<T> make_shared(size_t N, const remove_extent_t<T>& u); // \tcode{T} is \tcode{U[]}
   template<class T, class A>
-    shared_ptr<T> allocate_shared(const A& a, size_t N,
-                                  const remove_extent_t<T>& u);                 // \tcode{T} is \tcode{U[]}
+    constexpr shared_ptr<T> allocate_shared(const A& a, size_t N,
+                                            const remove_extent_t<T>& u);       // \tcode{T} is \tcode{U[]}
 
   template<class T>
-    shared_ptr<T> make_shared(const remove_extent_t<T>& u);                     // \tcode{T} is \tcode{U[N]}
+    constexpr shared_ptr<T> make_shared(const remove_extent_t<T>& u);           // \tcode{T} is \tcode{U[N]}
   template<class T, class A>
-    shared_ptr<T> allocate_shared(const A& a, const remove_extent_t<T>& u);     // \tcode{T} is \tcode{U[N]}
+    constexpr shared_ptr<T> allocate_shared(const A& a, const remove_extent_t<T>& u); // \tcode{T} is \tcode{U[N]}
 
   template<class T>
-    shared_ptr<T> make_shared_for_overwrite();                                  // \tcode{T} is not \tcode{U[]}
+    constexpr shared_ptr<T> make_shared_for_overwrite();                        // \tcode{T} is not \tcode{U[]}
   template<class T, class A>
-    shared_ptr<T> allocate_shared_for_overwrite(const A& a);                    // \tcode{T} is not \tcode{U[]}
+    constexpr shared_ptr<T> allocate_shared_for_overwrite(const A& a);          // \tcode{T} is not \tcode{U[]}
 
   template<class T>
-    shared_ptr<T> make_shared_for_overwrite(size_t N);                          // \tcode{T} is \tcode{U[]}
+    constexpr shared_ptr<T> make_shared_for_overwrite(size_t N);                // \tcode{T} is \tcode{U[]}
   template<class T, class A>
-    shared_ptr<T> allocate_shared_for_overwrite(const A& a, size_t N);          // \tcode{T} is \tcode{U[]}
+    constexpr shared_ptr<T> allocate_shared_for_overwrite(const A& a, size_t N);    // \tcode{T} is \tcode{U[]}
```

Comment:
If there is enough space for this comment, you may as well align all of the comments same as this.